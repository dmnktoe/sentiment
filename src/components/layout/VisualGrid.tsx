export default function VisualGrid() {
  return (
    <div className='fixed inset-0 z-[-1] mx-auto flex h-full justify-between px-2 sm:max-w-[var(--max-width)] sm:px-4'>
      {Array(5)
        .fill(0)
        .map((_, index) => (
          <div
            key={index}
            className={`h-full w-[1px] bg-grid ${index === 4 ? 'hidden sm:block' : ''}`}
          ></div>
        ))}
    </div>
  );
}
