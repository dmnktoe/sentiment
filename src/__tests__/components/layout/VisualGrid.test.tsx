import { render } from '@testing-library/react';
import React from 'react';

import VisualGrid from '@/components/layout/VisualGrid';

describe('VisualGrid', () => {
  it('renders a container div with correct classes', () => {
    const { container } = render(<VisualGrid />);
    const div = container.firstChild as HTMLElement;
    expect(div).toHaveClass(
      'fixed',
      'inset-0',
      'z-[-1]',
      'mx-auto',
      'flex',
      'h-full',
      'justify-between',
      'px-2',
    );
  });

  it('matches snapshot', () => {
    const { asFragment } = render(<VisualGrid />);
    expect(asFragment()).toMatchSnapshot();
  });
});
