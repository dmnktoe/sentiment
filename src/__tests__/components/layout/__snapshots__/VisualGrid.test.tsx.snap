// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`VisualGrid matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="fixed inset-0 z-[-1] mx-auto flex h-full justify-between px-2 sm:max-w-[var(--max-width)] sm:px-4"
  >
    <div
      class="h-full w-[1px] bg-grid "
    />
    <div
      class="h-full w-[1px] bg-grid "
    />
    <div
      class="h-full w-[1px] bg-grid "
    />
    <div
      class="h-full w-[1px] bg-grid "
    />
    <div
      class="h-full w-[1px] bg-grid hidden sm:block"
    />
  </div>
</DocumentFragment>
`;
